### PR TITLE
Add BUILD.md Development Guide

### LINKED ISSUE #1266: `make check` command to check if dependencies needed to build the project are installed.
I would like a make check command to see if I have the appropriate dependencies installed.

If this exists already, please refer me to the appropriate documentation and close this issue.

### PR DESCRIPTION
**Pull request by ihcsim**
## Change Overview

This PR adds the BUILD.md development guide, with instructions on how to test, build and package Kanister locally. It also adds a new `check` target to the Makefile to do some preliminary dev tools check.

```sh
✗ make check                                                                                                                           
* need Go....................... ⚠️
 → version mismatched - got 1.18, need 1.17
* need Docker................... ✅
* need Kubectl.................. ✅

✗ make check
* need Go....................... ✅
* need Docker................... ✅
* need Kubectl.................. ✅

✗ make check  
* need Go...................... ❌
 → Go not installed
* need Docker.................. ❌
 → Docker not installed
* need Kubectl................. ❌
 → Kubectl not installed

```

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [X] :world_map: Documentation
- [ ] :robot: Test

## Issues

- Fix #1266

### CODE REVIEW

File: BUILD.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+Use the `check` target to ensure your development environment has the necessary
+development tools:
+
+```sh
+make check
+```
+
+The following targets can be used to lint, test and build the Kanister
+controller:
```
Comment (viveksinghggits) [inline BUILD.md:43]: should we also add steps to build  `kanctl` and `kando` binaries. If not separately there should be a target that just builds all the binaries.

File: BUILD.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+## Architecture
+
+Kanister is a data management framework written in Go. It allows users to
+express data protection workflows using blueprints and actionsets. These
+resources are defined as Kubernetes
+[Custom Resource Definitions](https://docs.kanister.io/architecture.html#custom-resources)
+, following the operator pattern.
+
+![kanister workflow](./graphic/kanister_workflow.png)
+
```
Comment (viveksinghggits) [inline BUILD.md:14]: we can specify a link here from docs.kanister.io for more details.

**Review by viveksinghggits — COMMENTED** [review]
LGTM

File: BUILD.md (hunk 1 above)
Comment (pavannd1) [reply BUILD.md:43]: Can be done in a follow up PR 👍🏼

**Review by pavannd1 — APPROVED** [review]
LGTM

File: BUILD.md (hunk 1 above)
Comment (ihcsim) [reply BUILD.md:43]: I updated `build/build.sh` in https://github.com/kanisterio/kanister/pull/1355 to build `kanctl` and `kando`.

**Pull request merged**